### ISSUE TITLE
Federated calendar and contacts sharing

### ISSUE THREAD

**Issue by georgehrke** [nextcloud/server#1440]
There already is a federated file sharing, that allows you to share files among different Nextcloud instances.
We need something similar for calendars and address books.

suggested milestone: Nextcloud 12

**Comment by mimi89999** [comment]
Any workaround to get that?

**Comment by georgehrke** [comment]
well, you can use public calendar sharing and webcal feeds for read-only "federated" shares, but that's only Nextcloud11+ and will only work in the web interface

**Comment by georgehrke** [comment]
@schiessle Let's tackle this at the next hackweek, deal?

**Comment by schiessle** [comment]
@georgehrke deal! 😃

**Comment by MorrisJobke** [comment]
Seem to not have happened :/

**Comment by georgehrke** [comment]
@MorrisJobke we discussed it, but it was too complex for one week.

@schiessle Do you mind posting the notes we took? :)

**Comment by MorrisJobke** [comment]
Looks like it is included in #9345  ;) cc @schiessle

**Comment by MorrisJobke** [comment]
@schiessle What is the status of this? Doesn't seem to get into 14 anymore, right? move to 15?

**Comment by rugk** [comment]
Any news here?

**Comment by dev4223** [comment]
Hi there, would love to have this feature. Any status news? Thanks a lot.

**Comment by wargreen** [comment]
Hi, same here.
Thanks for who achieve it functionality !

**Comment by georgehrke** [comment]
Please use the [Github Reactions](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments) feature instead of commenting +1, 👍 etc. These kinds of comments just unnecessarily notify everyone subscribed to this issue. Thx! :)

**Comment by rugk** [comment]
@georgehrke That's true. However, after half a year of inactivity in an issue, it is very appropriate to ask whether there are any news, i.e. whether someone worked on it, or the issue has been forgotten.

Especially, as this was planned for Nextcloud 14 and right now v15 has already been released already.

**Comment by Eikir** [comment]
> @georgehrke That's true. However, after half a year of inactivity in an issue, it is very appropriate to ask whether there are any news, i.e. whether someone worked on it, or the issue has been forgotten.
> Especially, as this was planned for Nextcloud 14 and right now v15 has already been released already.

Sadly it's not an exception. What the people at Nextcloud Groupware are doing is non-transparent. It's not the only issue where they refuse to answer.
It's sad, because they've done and they are still doing tremendous work.

**Comment by georgehrke** [comment]
> Sadly it's not an exception. What the people at Nextcloud Groupware are doing is non-transparent. It's not the only issue where they refuse to answer.

![f9244c4e-a08a-4983-bd36-ed344887da88](https://user-images.githubusercontent.com/1250540/52897356-cbe2a700-31d3-11e9-83d2-e70a935bcc54.png)

There is nothing hidden here. As you can see in the right sidebar it's not assigned to any milestone.
For each version there is a feature planning, usually at the community hackweek. If we decide to implement this in the upcoming version, it will be assigned to the respecting milestone.

(And yes, obviously i could assign this to the 16 milestone, but i can already guarantee you that i won't have enough time to implement it and that people will complain if we assign it to a milestone and postpone it.)

**Comment by daraelmin** [comment]
Hi 

I've just created a bounty of $50 for this issue. If someone else is interested to raise it (or better to solve it), just follow the link below:
[https://www.bountysource.com/issues/37963414-federated-calendar-and-contacts-sharing]

Cheers!

**Comment by rootfuchs** [comment]
> Hi
> 
> I've just created a bounty of $50 for this issue. If someone else is interested to raise it (or better to solve it), just follow the link below:
> [https://www.bountysource.com/issues/37963414-federated-calendar-and-contacts-sharing]
> 
> Cheers!

You good Sir have my wallet!

**Comment by rugk** [comment]
Wow $150.00 USD for this now. Nice.

Would possibly be good if Nextcloud could add BugBounty integration here, so the money is displayed in the title etc.

**Comment by georgehrke** [comment]
> Would possibly be good if Nextcloud could add BugBounty integration here, so the money is displayed in the title etc.

There should be one in place already ...
@jospoortvliet

**Comment by rugk** [comment]
Well… it is not displayed :thinking:

**Comment by strypey** [comment]
I'm wondering if the federated calendar features envisioned for NextCloud will only federate between NC instances, or whether there will be interop with other events calendar sites running software that implements ActivityPub? [Friendica events federate over AP](https://socialhub.activitypub.rocks/t/activitypub-at-36c3-the-decentral-assembly-and-fediverse-party/402/5) already, there is at least one instance of [Gath.io that federates events over AP](http://events.friend.camp/), and [Mobilizon has a beta test site](https://test.mobilizon.org/) up now (will eventually federate over AP). It would be great to have interop between all these services and any others that support AP (GetTogether, Gancio etc), and there's been an ongoing discussion about how to achieve this here: 
https://socialhub.activitypub.rocks/t/federated-events/305/16

**Comment by genofire** [comment]
Is it possible to implement sharing/federation with a group on another nextcloud?

**Comment by chaptergy** [comment]
Officially federating calendars is implemented via the [Circles App](https://github.com/nextcloud/circles) using federated circles, however in practice it has never worked for me. I'm trying it periodically, however it has not been fixed yet. (See issue https://github.com/nextcloud/circles/issues/254)

**Comment by genofire** [comment]
My fault found the group settings in the global settings of the nextcloud:
- `Allow users on this server to receive group shares from other servers`
- `Allow users on this server to send shares to groups on other servers`

**Comment by daraelmin** [comment]
Maybe this pull request will solve this issue: #9345 

It should allow the implementation of federated calendar and contact sharing.

**Comment by georgehrke** [comment]
Not really. What will actually bring this forward is https://github.com/nextcloud/server/pull/20214

**Comment by daraelmin** [comment]
@rubo77  bounty increased to 250 dollars
Could you edit the issue title please ?

**Comment by rubo77** [comment]
done.

I thought, we should not use boutysource any more?

**Comment by daraelmin** [comment]
Thanks.

They gave up changing their condition in front of the reaction of the community, so....

**Comment by florom** [comment]
Hi All,

I would appreciate a login to another calendar (iCloud or Google) nativly in the calendar app to cause the issue came up within my family and I could not manage to switch to (my) nextcloud.
Since public calendars are per definition public I do not want to use them although it is very easy and the synchronisation interval can be changed easily.

I use a workaround for this issue using Outlook and the Addon CalDavSynchronizer.

I made a separate calendar Outlook for synchronisation.
With one profile I connect to iCloud using my login and password and with the other profil I connect to nextcloud using login and password.
So every time I start Outlook (which I use for emails so I start it at least once a day) iCloud and Nextcloud calendar are syncronising.

I am reasarching for a calendar program on linux (server) which can be run permanently as proxy(which is now Outlook) but I have not done much in this direction. I will give feedback if I find a solution. 

Hope this helps a few people.

**Comment by sunjam** [comment]
That has nothing to do with this issue about federation of calendars and
contacts

**Comment by florom** [comment]
Hi @sunjam ,

I agree with you that its not about federation but it is about sharing.

in 
https://github.com/nextcloud/calendar/issues/3318#issuecomment-878013204
@ChristophWurst states that the other issue sounds like a duplicate to this one.

Since it is no good to open up new issues (or comment in different issues) on the same topic
I used this one.
If the decision will be changed I post the workaround in the other Issue. 
That is fine with me.

**Comment by sunjam** [comment]
It is good to open new issues.  Please do so so we can keep this related to
the actual topic. just my .02

**Comment by florom** [comment]
Hi @sunjam ,

I am not considering as new:

> but it is about sharing

so the issue has the correct  topic.

**Comment by Booteille** [comment]
Hi.

Any plan to move on this topic anytime soon?

**Comment by rhinozeross** [comment]
Hey Guys! Sorry for asking this stupid question but i didn't find anything in the Internet yet:/ how can i get a public Calender i created with User A on Instance A to a User B on Instance B? i tried to share it via federation but i could not see user B on Instance A thought both instances are conneted via Federation. i also couldn't import it vial Calendar-> import.. both instances run on Version 22.2.6 and will get an upgrade to 23 soon..

**Comment by labor4** [comment]
AFAIK there is subscription for CalDAV, but not für CardDAV.
Meaning read-only Calendar is possible, but not for Contacts.

First having **subscriptions for Contacts** to may be worth considering as an **intermediate goal**?

Use case: In some situations I think it preferable to separate an NC instance with calendar/contacts from a heavy lifting file-sharing NC instance. The file-NC could benefit from the contacts inventary while not strictly needing to have write permissions.

**Comment by Spacelord09** [comment]
Are there any updated to federated calendar sharing?

**Comment by st3iny** [comment]
> Are there any updates to federated calendar sharing?

Federated calendar sharing is currently being developed. However, only the read-only part for now.

**Comment by einlichtvogel** [comment]
are there any updates on this topic?

**Comment by szaimen** [comment]
Work is going on in https://github.com/nextcloud/server/pull/58008
